hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -168,10 +168,12 @@ private TokenAssociation validateAndBuildAutoAssociation(
         final var tokenRel = tokenRelStore.get(accountId, tokenId);
         final var config = handleContext.configuration();
 
-        if (tokenRel == null && account.maxAutoAssociations() > 0) {
-            validateFalse(
-                    account.usedAutoAssociations() >= account.maxAutoAssociations(),
-                    NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
+        if (tokenRel == null && (account.maxAutoAssociations() > 0 || account.maxAutoAssociations() == -1)) {
+            if (account.maxAutoAssociations() != -1) {
```

Comment:
do you not care about `entitiesConfig.unlimitedAutoAssociationsEnabled()` here?

Reply:
This might be a good use for the method I suggested [in another comment](https://github.com/hashgraph/hedera-services/pull/13429/files#r1622973066)

Reply:
Yes It seems that here we also care about the feature flag. I'm now using the method call suggested by Joseph.